docs/openapi.yaml

```diff
@@ -0,0 +1,760 @@
+        404:
+          description: no resource exists with the given `resourcePath`
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NotFound'
+    post:
+      tags:
+        - resource
+      description: >-
+        Add a new subresource, *underneath* an existing resource. The path *is
+        allowed to be empty*, in which case `resourcePath` is empty, the resource
+        is created under the root resource, as `/new_resource`.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Resource'
+      responses:
+        201:
+          description: JSON representation of successfully-created resource
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  created:
+                    $ref: '#/components/schemas/Resource'
+        404:
+          description: no resource exists with the given `resourcePath`
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/UserError'
+    put:
+      tags:
+        - resource
+      description: >-
+        Overwrite an existing resource. This endpoint requires a fully-formed
+        resource model (and cannot patch over individual fields on the existing
+        resources).
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Resource'
+      responses:
+        200:
+          description: JSON representation of successfully-updated resource
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  updated:
+                    $ref: '#/components/schemas/Resource'
+        400:
+          description: invalid input (missing fields or fields have incorrect types)
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/UserError'
+        404:
+          description: no resource exists with the given `resourcePath`
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NotFound'
+    delete:
+      tags:
+        - resource
+      responses:
+        204:
+          description: resource successfully deleted
+        404:
+          description: no resource exists with the given `resourcePath`
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NotFound'
+  /role/:
+    get:
+      tags:
+        - role
+      description: List all the roles registered in arborist.
+      responses:
+        200:
+          description: Success
+          content:
+            application/json:
+              # example JSON:
```

Comment:
dead code, could remove unless there's a valid reason to keep it.

Reply:
will remove